app/controllers/invoices_controller.rb

```diff
@@ -0,0 +1,112 @@
+#    Copyright 2020 Operation Paws for Homes
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+class InvoicesController < ApplicationController
+
+  PER_PAGE = 30
+
+  before_action :require_login, except: %i[show]
+  before_action :unlocked_user, except: %i[show]
+  before_action :select_bootstrap41
+  before_action :show_user_navbar, except: %i[show]
+
+  def new
+    load_invoiceable
+    @invoice = Invoice.new
+  end
+
+  def create
+    load_invoiceable
+    @invoice = @invoiceable.invoices.build(invoice_create_params)
+    @invoice.slug = SecureRandom.urlsafe_base64(32)
+    @invoice.user_id = current_user.id
+    @invoice.status = 'open'
+    if @invoice.save
+      flash[:success] = "Invoice Generated"
+      redirect_to polymorphic_path([@invoiceable.adopter])
+    else
+      flash.now[:error] = "Invoice could not be saved"
+      render 'new'
+    end
+  end
+
+  def show
+    @invoice = Invoice.friendly.find(params[:id])
+  end
+
+  def edit
+  end
+
+  def update
+    @invoice = Invoice.friendly.find(params[:id])
+    @invoice.card_token =  stripe_params["stripeToken"]
+
+    begin
+      @invoice.process_payment(@invoice.invoiceable.adopter.email)
+    rescue Stripe::CardError => e
+      flash[:error] = e.message
+    else
+      @invoice.donation = stripe_params["invoice"]["donation"]
```

Comment:
get this wrapped in a transaction since you have 2 saves going on, I think you'd want `save!` too so something explodes if either goes awry, try that and if those are ok then send the email, I can see possibly sending an email to someone when something explodes